backend/src/test/java/de/wolffclan/backend/services/date/DateServiceTest.java

```diff
@@ -0,0 +1,275 @@
+package de.wolffclan.backend.services.date;
+
+import de.wolffclan.backend.models.date.Date;
+import de.wolffclan.backend.models.date.NewDate;
+import de.wolffclan.backend.repositories.date.DateRepository;
+import de.wolffclan.backend.repositories.user.UserRepository;
+import org.junit.jupiter.api.Test;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+class DateServiceTest {
+    DateRepository dateRepository = mock(DateRepository.class);
+    UserRepository userRepository = mock(UserRepository.class);
+    DateService dateService = new DateService(dateRepository, userRepository);
+
+    @Test
+    void getAllDates() {
+        // GIVEN
+        Date date = createDate();
+        List<Date> expected = new ArrayList<>(List.of(date));
+        // WHEN
+        when(dateRepository.findAll()).thenReturn(expected);
+        List<Date> actual = dateService.getAllDates();
+        verify(dateRepository).findAll();
+        assertEquals(expected, actual);
```

Comment:
Zeile 29: Ich hab bei mir immer das `expected` in den "Then"-Block gepackt. Der könnte bei Dir übrigens etwa vor Zeile 33 anfangen.
Zeile 31: Das `when` gehört aber m.M.n., obwohls so heißt, eigentlich eher in den "Given"-Block, weil's den zu testenden Aufruf vorbereitet, der dann mehr oder weniger allein im "When"-Block steht.

Genug der Klugschieterei

Wie ich weiter unten sehe, behältst Du ein Schema wenigstens bei, was schonmal viel wert ist.

Reply:
Nee, zumindest nach dem, wie ich die 3 Teile interpretiere, nich.
_**"Given"**_ sind die Vorgaben, die ich brauche um die zu testende Methode auszuführen. (Konstanten, Mocking-Definitionen, ...)
**_"When"_** ist dann der eigentliche Methoden-Aufruf
Und **_"Then"_** dann die Verarbeitung der Ergebnisse. Hier werden natürlich einige Konstanten aus "Given" wierderholt, entweder als Variable oder direkt als Wert.